consul-lambda-registrator/trigger.go

```diff
@@ -132,21 +142,15 @@ func (e Environment) GetLambdaData(ctx context.Context, arn string) ([]Event, er
 		}
 	}
 
-	var em *EnterpriseMeta
-	if v, ok := tags[namespaceTag]; ok {
-		em = &EnterpriseMeta{Namespace: v, Partition: "default"}
-	}
-
-	if v, ok := tags[partitionTag]; ok {
-		if em == nil {
-			em = &EnterpriseMeta{Namespace: "default", Partition: v}
-		} else {
-			em.Partition = v
-		}
+	// Get enterprise metadata from the tags. This will be nil for OSS.
+	em := structs.NewEnterpriseMeta(tags[partitionTag], tags[namespaceTag])
+	if !e.IsEnterprise && em != nil {
+		return nil, errNotEnterprise
 	}
 
-	if !e.IsEnterprise && em != nil {
-		return nil, notEnterpriseErr
+	if e.IsEnterprise && em == nil {
+		// If enterprise but no tags were provided, then use the default AP and NS.
+		em = structs.NewEnterpriseMeta("default", "default")
```

Comment:
Won't this be defaulted on on line 146?

Reply:
The way the code is setups, for OSS `EnterpriseMeta` is always set to `nil`. So if `NewEnterpriseMeta` is called with an empty `partition` and `namespace` it returns `nil`. This `if` statement ensures that when we are using enterpise and both the `partition` and `namespace` tags are empty, we init the `em` field with `default/default` so it is not `nil` and we know that we're dealing with enterprise.

It's a little bit clunky in this spot but it makes the rest of the event handling code more streamline.